tests/integration/test_design_export.py

```diff
@@ -0,0 +1,250 @@
+
+    # Check the exported file
+    assert file_location.exists()
+
+    # Import the scdocx
+    design_read = modeler.open_file(file_location)
+
+    # Check the imported design
+    _checker_method(design_read, design, True)
+
+
+def test_export_to_parasolid_text(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
+    """Test exporting a design to parasolid text format."""
+    # Create a demo design
+    design = _create_demo_design(modeler)
+
+    # Define the location and expected file location
+    location = tmp_path_factory.mktemp("test_export_to_parasolid_text")
+
+    if modeler.client.backend_type == BackendType.LINUX_SERVICE:
+        file_location = location / f"{design.name}.x_t"
+    else:
+        file_location = location / f"{design.name}.xmt_txt"
+
+    # Export to parasolid text
+    design.export_to_parasolid_text(location)
+
+    # Check the exported file
+    assert file_location.exists()
+
+    # TODO: Check the exported file content
+
+
+def test_export_to_parasolid_binary(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
+    """Test exporting a design to parasolid binary format."""
+    # Create a demo design
+    design = _create_demo_design(modeler)
+
+    # Define the location and expected file location
+    location = tmp_path_factory.mktemp("test_export_to_parasolid_binary")
+
+    if modeler.client.backend_type == BackendType.LINUX_SERVICE:
+        file_location = location / f"{design.name}.x_b"
+    else:
+        file_location = location / f"{design.name}.xmt_bin"
+
+    # Export to parasolid binary
+    design.export_to_parasolid_bin(location)
+
+    # Check the exported file
+    assert file_location.exists()
+
+    # TODO: Check the exported file content
+
+
+def test_export_to_step(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
+    """Test exporting a design to STEP format."""
+    # Create a demo design
+    design = _create_demo_design(modeler)
+
+    # Define the location and expected file location
+    location = tmp_path_factory.mktemp("test_export_to_step")
+    file_location = location / f"{design.name}.stp"
+
+    # Export to step
+    design.export_to_step(location)
+
+    # Check the exported file
+    assert file_location.exists()
+
+    # Import the scdocx
+    design_read = modeler.open_file(file_location)
+
+    # Check the imported design
+    _checker_method(design_read, design, False)
+
+
+def test_export_to_iges(modeler: Modeler, tmp_path_factory: pytest.TempPathFactory):
+    """Test exporting a design to IGES format."""
+    # Create a demo design
+    design = _create_demo_design(modeler)
+
+    # Define the location and expected file location
+    location = tmp_path_factory.mktemp("test_export_to_stl")
+    file_location = location / f"{design.name}.igs"
+
+    # Export to IGES
+    design.export_to_iges(location)
+
+    # Check the exported file
+    assert file_location.exists()
+
+    # TODO: Check the exported file content
```

Comment:
Why the TODOs? Can't these files be read by the client?

Reply:
The service has a problem when reading, but not when exporting. I opened up #1146 for this reason 😢